addons/website/models/mixins.py

```diff
@@ -252,3 +255,100 @@ def _search_website_published(self, operator, value):
             return (['!'] if value is False else []) + expression.AND([is_published, on_current_website])
         else:  # should be in the backend, return things that are published anywhere
             return is_published
+
+
+class WebsiteSearchableMixin(models.AbstractModel):
+    """Mixin to be inherited by all models that need to searchable through website"""
+    _name = 'website.searchable.mixin'
+    _description = 'Website Searchable Mixin'
+
+    @api.model
+    def _search_build_domain(self, domain, search, fields, extra=None):
+        """
+        Builds a search domain AND-combining a base domain with partial matches of each term in
+        the search expression in any of the fields.
+
+        :param domain: base domain combined in the search expression
+        :param search: search expression string
+        :param fields: list of field names to match the terms of the search expression with
+        :param extra: function that returns an additional subdomain for a search term
+
+        :return: domain limited to the matches of the search expression
+        """
+        domains = domain.copy()
+        if search:
+            for search_term in search.split(' '):
+                subdomains = []
+                for field in fields:
+                    subdomains.append([(field, 'ilike', escape_psql(search_term))])
+                if extra:
+                    subdomains.append(extra(self.env, search_term))
+                domains.append(expression.OR(subdomains))
+        return expression.AND(domains)
```

Comment:
That doesn't seem correct, assuming the input `domain` is an actual domain. This treats the input domain as a list of domains.

Reply:
Indeed, the signature is misleading. Replacing the `domain` parameter by `domain_list`.